testing/cli/test_cli_sct_compute_compression.py

```diff
@@ -0,0 +1,155 @@
+                                           '-sex', 'J', '-o', filename, '-normalize-hc',  '1'])
+        assert e.value.code == 2
+
+
+def test_sct_compute_compression_check_wrong_age(tmp_path, dummy_3d_mask_nib, dummy_3d_compression_label, dummy_3d_vert_label):
+    """ Run sct_compute_compression with wrong age"""
+    filename = str(tmp_path / 'tmp_file_out.csv')
+    with pytest.raises(SystemExit) as e:
+        sct_compute_compression.main(argv=['-i', dummy_3d_mask_nib, '-l', dummy_3d_compression_label, '-vertfile', dummy_3d_vert_label,
+                                           '-age', '20', '-o', filename, '-normalize-hc',  '1'])
+        assert e.value.code == 2
+
+
+def test_sct_compute_compression_check_wrong_metric(tmp_path, dummy_3d_mask_nib, dummy_3d_compression_label, dummy_3d_vert_label):
+    """ Run sct_compute_compression when specifying a wrong metric"""
+    filename = str(tmp_path / 'tmp_file_out.csv')
+    with pytest.raises(SystemExit) as e:
+        sct_compute_compression.main(argv=['-i', dummy_3d_mask_nib, '-l', dummy_3d_compression_label, '-vertfile', dummy_3d_vert_label,
+                                           '-metric', 'MEAN', '20', '-o', filename, '-normalize-hc',  '0'])
+        assert e.value.code == 2
+
+
+def test_sct_compute_compression_no_normalization(tmp_path, dummy_3d_mask_nib, dummy_3d_compression_label, dummy_3d_vert_label):
+    """ Run sct_compute_compression without normalization to a database of healthy controls"""
+    filename = str(tmp_path / 'tmp_file_out.csv')
+    sct_compute_compression.main(argv=['-i', dummy_3d_mask_nib, '-l', dummy_3d_compression_label, '-vertfile', dummy_3d_vert_label,
+                                       '-normalize-hc',  '0', '-o', filename])
+    with open(filename, "r") as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=',')
+        row = next(reader)
+        assert float(row['compression_level']) == 5.0
+        assert float(row['diameter_AP_ratio']) == pytest.approx(20.040803711692355)
+        assert row['normalized_diameter_AP_ratio'] == ''
+
+
+def test_sct_compute_compression(tmp_path, dummy_3d_mask_nib, dummy_3d_compression_label, dummy_3d_vert_label):
+    """ Run sct_compute_compression and check mscc and normalized mscc"""
+    filename = str(tmp_path / 'tmp_file_out.csv')
+    sct_compute_compression.main(argv=['-i', dummy_3d_mask_nib, '-l', dummy_3d_compression_label,
+                                       '-vertfile', dummy_3d_vert_label, '-normalize-hc',  '1', '-o', filename])
+    with open(filename, "r") as csvfile:
+        reader = csv.DictReader(csvfile, delimiter=',')
+        row = next(reader)
+        assert float(row['compression_level']) == 5.0
+        assert float(row['diameter_AP_ratio']) == pytest.approx(12.525502319807725)
+        assert float(row['normalized_diameter_AP_ratio']) == pytest.approx(16.985020560800656)
```

Comment:
I notice that the `diameter_AP_ratio` is different between these two tests. This is because, in the `-normalize-hc 1` case, we use the metrics computed from `sct_process_segmentation -normalize-PAM50` instead:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/246d97b5e92621f703749763790841383a0acc8f/spinalcordtoolbox/scripts/sct_compute_compression.py#L627

Using the same name (`diameter_AP_ratio`) in both cases is a bit confusing, because `diameter_AP_ratio` implies that no normalization occurs. But, in fact, for `-normalize-hc 1`, `diameter_AP_ratio` is normalized to the PAM50 template, making it different that the same-name `diameter_AP_ratio` from `-normalize-hc 0`. This might create a trap for users, who don't realize that the two quantities are different.

I think it might be good to find a way to communicate this distinction to the user. For example:

- Have 3 keys instead of 2: `diameter_AP_ratio`, `diameter_AP_ratio_normalized_PAM50`, and `diameter_AP_ratio_normalized_HC`
  - `diameter_AP_ratio` is the computed same way regardless of whether `-normalize-hc` is 0 or 1.
  - The two `normalized` metrics are only computed for `-normalize-hc 1`.
- Then, add in the description for `-normalize-hc` that two forms of normalization are performed (PAM50 template normalization, PAM50 + HC database normalization).

Reply:
This would require a bit of refactoring, and I understand that @sandrinebedard and @valosekj have ISMRM right now and are going on vacation soon, so if this is too much to complete right now, I'm happy to take care of this. (Only if you agree with the suggestion, of course!)

Reply:
This is an excellent point!
When processing the `sci-zurich` dataset, I manually set the output filename using the `-o` flag to make it clear whether `-normalize-hc` was used. See [these lines](https://github.com/sct-pipeline/dcm-metric-normalization/blob/c44dd1f8549766df6e02d2ed2cbe36eef79c678f/scripts/process_data_dcm-zurich.sh#L195-L196). But definitely, using dedicated keys would make it cleaner.

I am just thinking about the keys' names. Maybe I would go with the following:
- `diameter_AP_ratio` - obtained using `-normalize-hc 0` -- metrics computed in the native space without normalization
- `diameter_AP_ratio_PAM50` - obtained using `-normalize-hc 1` -- metrics computed in the PAM50 space without normalization
- `diameter_AP_ratio_PAM50_normalized` - obtained using `-normalize-hc 1` -- metrics computed in the PAM50 space with normalization to HC database

But I would also wait for @sandrinebedard opinion.

Reply:
> We could always begin the refactoring with your proposed names? Then, if there is further discussion on what the names should be, it would be a simple update, since it is just a find and replace.

I'm going to try and tackle this later today, then push to this PR. If there are any issues/concerns, we can always revert the commits or modify them. But, still, I'm just happy to try to lessen the menial work on you right now, since this is kind of boring refactoring work, hehe. :)

Reply:
Done!

I've tried my best to break each change down into individual commits, that way every bit of refactoring is straightforward and easy to understand. You should be able to look through each commit, one at a time, to understand what I've done. :slightly_smiling_face:

Reply:
Thank you very much for this refactoring, @joshuacwnewton! I also really appreciate your detailed commit messages making the code review very convenient!

I tested your changes, and their work perfectly!

The command I used:
```
sct_compute_compression -i sub-260155_acq-axial_T2w_label-SC_mask.nii.gz  -vertfile sub-260155_acq-axial_T2w_label-SC_mask_labeled.nii.gz  -l sub-260155_acq-axial_T2w_label-compression-manual.nii.gz  -normalize-hc 1 -o sub-260155_acq-axial_T2w_diameter_AP_normalize-hc.csv
```

The output CSV file looks like:

<img width="1016" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/8f23cd14-8762-456f-8a7f-0f9e4bece0a6">

I have just one minor comment here. The last column is now named `diameter_AP_ratio_normalized`. But maybe, we could call it `diameter_AP_ratio_PAM50_normalized` (as I mentioned in 
https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003#discussion_r1234325392) to make explicitely clear that the normalization is done on metrics computed in PAM50 space.

Reply:
Oh, oops! That was definitely an accidental mistake on my part. Thank you for catching that!

Reply:
Re: @valosekj's https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003#discussion_r1234325392:

> But I would also wait for @sandrinebedard opinion.

Do you have any thoughts about the this refactor, @sandrinebedard? Anything you object to, or that should be changed?

If there are no objections, then I think this PR might be ready to merge. :slightly_smiling_face:

Reply:
everything looks good to me!